src/synth/fluid_synth.c

```diff
@@ -5667,35 +5931,48 @@ fluid_synth_set_chorus_full(fluid_synth_t *synth, int set, int nr, double level,
     /* if non of the flags is set, fail */
     fluid_return_val_if_fail(set & FLUID_CHORUS_SET_ALL, FLUID_FAILED);
 
+    /* fx unit shadow values are set here so that they will be returned if queried */
     /* Synth shadow values are set here so that they will be returned if queried */
     fluid_synth_api_enter(synth);
 
-    if(set & FLUID_CHORUS_SET_NR)
+    ret = fluid_rvoice_mixer_set_chorus_full(synth->eventhandler->mixer, fxunit_idx,
+                                             set, nr, level, speed, depth_ms, type);
+    if(ret == FLUID_FAILED)
     {
-        synth->chorus_nr = nr;
+        FLUID_API_RETURN(FLUID_FAILED);
     }
 
-    if(set & FLUID_CHORUS_SET_LEVEL)
+    if (fxunit_idx < 0)
     {
-        synth->chorus_level = level;
-    }
+        /* Synth shadow values are set here so that they will be returned if queried */
+        if(set & FLUID_CHORUS_SET_NR)
+        {
+            synth->chorus_nr = nr;
+        }
 
-    if(set & FLUID_CHORUS_SET_SPEED)
-    {
-        synth->chorus_speed = speed;
-    }
+        if(set & FLUID_CHORUS_SET_LEVEL)
+        {
+            synth->chorus_level = level;
+        }
 
-    if(set & FLUID_CHORUS_SET_DEPTH)
-    {
-        synth->chorus_depth = depth_ms;
-    }
+        if(set & FLUID_CHORUS_SET_SPEED)
+        {
+            synth->chorus_speed = speed;
+        }
 
-    if(set & FLUID_CHORUS_SET_TYPE)
-    {
-        synth->chorus_type = type;
+        if(set & FLUID_CHORUS_SET_DEPTH)
+        {
+            synth->chorus_depth = depth_ms;
+        }
+
+        if(set & FLUID_CHORUS_SET_TYPE)
+        {
+            synth->chorus_type = type;
+        }
     }
 
-    param[0].i = set;
+    /* put fx unit index in bits [b15..b8] and set mask in bits [b7..b0] */
+    param[0].i = (fxunit_idx << 8) | (set & FLUID_CHORUS_SET_ALL) ;
```

Comment:
Be careful with signed shifts: https://sonarcloud.io/project/issues?id=FluidSynth_fluidsynth&open=AXVgLmC8i5dl9RN5Jo-q&pullRequest=673&resolved=false&types=BUG

I would prefer to store both parameters separately.